lib/lotus/commands/new.rb

```diff
@@ -10,11 +11,11 @@ class New
       attr_reader :app_name, :source, :target, :cli, :options
 
       def initialize(app_name, environment, cli)
-        @app_name = app_name
+        @app_name = ApplicationName.new(app_name)
         @options  = environment.to_options
         @arch     = @options.fetch(:architecture)
 
-        @target   = Pathname.pwd.join(@app_name)
+        @target   = Pathname.pwd.join(@app_name.to_s)
```

Comment:
I'd say we should decouple string representation (`#to_s`), from file system's manipulations. What do you think if we introduce `ApplicationName#to_path`?